### DISCUSSION TITLE
regarding bimodal microstructure

### DISCUSSION THREAD

**Discussion by anujyadav0693** [neperfepx/neper#868] [Q&A] [answered]
respected sir,
i am using this code
neper -T -n from_morpho -morpho "diameq:lognormal(0.0005,0.00005)+0.1111111111*lognormal(0.00001,0.0000001),1-sphericity:lognormal(0.145,0.03)" -group "diameq>0.00002?1:2" -morphooptiini "weight:radeq,coo:packing" -dim 2 -domain "square(0.04,0.01)" -o void -format "geo,ori,tess"  -statcell diameq -oridescriptor "euler-bunge" -ori "fiber(0,0,1,0,0,1)" -statgroup area -statface id,area

To generate a bimodal microstructure, but i have a query
is there any way to control the small grain percentage.
for example- i want to make multiple models as give in table below-

**_small grain size_**                                             **_small grain percentage_**
                                                    model1                       model2              model3               model4

50 micrometer                             10%                            20%                         30%                  50%

30 micrometer                              10%                             20%                           30%                   50%

My question is, in any way,  can I make multiple models of the same small grain dia with different small grain percentages or vice versa?
because I also want to make models of the same small grain percentages with different small grain diameter 
Is there any command to control, or should I do the hit-and-trial
thank you.

**Comment by anujyadav0693** [comment]
If possible, please update the given code.
Thankyou.

**Comment by rquey** [comment, marked as answer]
Neper's `-morpho` works with number fractions of grains, not volume fractions. You can reasonably convert one into the other using the average grain sizes.

**Comment by anujyadav0693** [comment]
neper -T -n from_morpho -morpho "diameq:0.3*lognormal(0.0005,0.00005)+lognormal(0.00005,0.000005),1-sphericity:lognormal(0.145,0.03)" -group "diameq>0.00009?1:2" -morphooptiini "weight:radeq,coo:packing" -dim 2 -domain "square(0.03,0.01)" -o void -format "geo,ori,tess"  -statcell diameq -oridescriptor "euler-bunge" -ori "fiber(0,0,1,0,0,1)" -statgroup area -statface id,area 




sir, I am using this still I am getting 3.75% of area following the small grain distribution, and the remaining area is given to large grain. This also take to much time to run. please suggest what type of formula this code follows. any way to reduce time
